_posts/2024-11-17-phpstan-performance-on-different-hardware.md

```diff
@@ -0,0 +1,71 @@
+---
+tags:
+    - PHPStan
+
+image: "images/og-images/phpstan-performance-on-different-hardware.jpg"
+
+ogImage:
+    title: "How much impact has hardware on PHPStan performance?"
+    imageUrl: "https://staabm.github.io/staabm.svg"
+    fileName: "phpstan-performance-on-different-hardware"
+---
+
+
+I recently updated my macbook to the newly released model.
+The idea was to get a faster feedback loop when working on complex stuff like PHPStan itself.
+
+### How much impact has hardware on PHPStan performance?
+
+Let's find out what we can expect when running different hardware.
+
+For the sake of this article we compare running [PHPStan@cc4eb92](https://github.com/phpstan/phpstan-src/commit/cc4eb92285fd8c96e595437cb9c593553bb5e957).
+We are running `time make` from within the terminal/console and the project root 5 times. We do so after a fresh boot-up and without any other applications running.
+These numbers are not scientific, but give us a rough idea.
+
+#### Running macOS
+
+```
+$ php -v
+PHP 8.3.13 (cli) (built: Oct 22 2024 18:39:14) (NTS)
+Copyright (c) The PHP Group
+Zend Engine v4.3.13, Copyright (c) Zend Technologies
+```
+
+Apple MacBook M1 Pro (2021), 10‑Core CPU, 1 TB SSD, 32 GB RAM (on full battery)
+- 77-85 seconds
+
+Apple MacBook M4 Pro (2024), 14‑Core CPU, 1 TB SSD, 48 GB RAM (on full battery)
```

Comment:
I think I would put the M4 after the M2 test. It looks a bit confusing to me personally.

M1, M4 and then M2? 🤔

Reply:
I put things together which have the same php-version/setup.

on your notebook there was opcache involved